### ISSUE TITLE
Feature: CloudStack detach ISO

### ISSUE THREAD

**Issue by andrijapanicsb** [hashicorp/packer#7235]
Packer version: 1.3.3
Linux 64 (Centos 7.5 - not relevant)
Debug log - not relevant (read later why)
Builder - CloudStack
KVM hypervisor

When building template using ISO file and kickstart (i.e. centos7 template build using ISO and ks), user want to eject CDROM (detach ISO from CloudStack) at the end of the installation, so the OS can boot from local hardrive/volume and then packer can proceed with SSH connection etc... (this is done inside guest OS as part of kickstart for Centos7)...but...

Problem: On KVM this is not working/possible inside the Guest OS (simply doesn't work...), so we need to handle it vie API to CloudStack, to detach the ISO at some moment (similar happens with VMware @ CLoudStack, but here VMware still boots from local HDD as needed - so it's OK for VMware by accident).

I would propose making a configurable timeout (from the beginning of builder execution), to detach the ISO, or some perhaps better way ? - here you can check how it was done in a fork (a bit outdated): https://github.com/schubergphilis/packer-cloudstack

Idea is when OS install is done inside VM, that on reboot the VM should boot from it's local HDD, instead of again from ISO file (endless loop).

**Comment by SwampDragons** [comment]
Can you check whether the build [here](https://circleci.com/gh/hashicorp/packer/12233#artifacts/containers/0) solves this issue for you? explanation for template options to use can be found in the PR #8106

**Comment by andrijapanicsb** [comment]
Yep, a colleague of mine did that commit  - all good. We can close this one.

**Comment by SwampDragons** [comment]
hah, awesome.
